rewrite-toml/src/test/java/org/openrewrite/toml/TomlParserTest.java

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.toml;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.toml.Assertions.toml;
+
+class TomlParserTest implements RewriteTest {
+    @Test
+    void keyValueString() {
+        rewriteRun(
+          toml(
+            """
+            str = "I'm a string. \\"You can quote me\\". Name\\tJos\\u00E9\\nLocation\\tSF."
+            """
+          )
+        );
+    }
+
+    @Test
+    void keyValueInteger() {
+        rewriteRun(
+          toml(
+            """
+            int1 = +99
+            int2 = 42
+            int3 = 0
+            int4 = -17
+            int5 = 1_000
+
+            # hexadecimal with prefix `0x`
+            hex1 = 0xDEADBEEF
```

Comment:
Ah the classic `0xDEADBEEF` I was so baffled/amused the first time I saw that value while debugging